USAGOV-2240: PXBF Release v0.10.0.beta.1

scottqueen-bixal opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
PXBF Release v.0.10.beta.1

https://github.com/orgs/GSA/projects/48/views/17?filterQuery=-no%3Amilestone+repo%3A%22GSA%2Fpx-benefit-finder%22+milestone%3A%22v0.10.0.beta.1%22

<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-
https://github.com/GSA/px-benefit-finder/milestone/17

## Description
<!--- Summarize the changes made in this pull request, not what it's for. -->

### Application Improvements
[Nested criteria not open as expected](https://github.com/GSA/px-benefit-finder/issues/2086)	https://github.com/GSA/px-benefit-finder/issues/2086
[Clicking Back buttons on Firefox browser after entering required fields on ‘death of a loved one’ add an extra route with # on ‘about-the-deceased’ route](https://github.com/GSA/px-benefit-finder/issues/2058)	https://github.com/GSA/px-benefit-finder/issues/2058
[child criteria is required cause error](https://github.com/GSA/px-benefit-finder/issues/2089)	https://github.com/GSA/px-benefit-finder/issues/2089

### Custom Module Improvements
[Remove boolean type of criteria](https://github.com/GSA/px-benefit-finder/issues/2080)	https://github.com/GSA/px-benefit-finder/issues/2080
[Life event form criteria error message not showing in JSON data](https://github.com/GSA/px-benefit-finder/issues/2088)	https://github.com/GSA/px-benefit-finder/issues/2088
[PHPStan error fixing](https://github.com/GSA/px-benefit-finder/issues/2039)	https://github.com/GSA/px-benefit-finder/issues/2039

### a11y Improvements
[Skip to main is breaking on tab enter](https://github.com/GSA/px-benefit-finder/issues/2065)	https://github.com/GSA/px-benefit-finder/issues/2065
[Announce Web Page Title Changes to Screen Reader Users](https://github.com/GSA/px-benefit-finder/issues/1904)	https://github.com/GSA/px-benefit-finder/issues/1904

### design Improvements
[Color Updates from USA.gov AUDIT](https://github.com/GSA/px-benefit-finder/issues/2073)	https://github.com/GSA/px-benefit-finder/issues/2073

## Type of Changes
<!--- Put an `x` in all the boxes that apply. -->
- [ ] New Feature
- [x] Bugfix
- [x] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [x] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] WWW
  - [ ] Egress
  - [ ] Tools
  - [ ] Cron
- [ ] Other

## Testing Instructions
<!-- This instructions are different from “testing instructions” in Jira – those are typically for Content/UX stakeholders -->
<!-- Not “see Jira” – if they are really the same, copy and paste. -->
### Application Improvements
[Nested criteria not open as expected, under Date field types](https://github.com/GSA/px-benefit-finder/issues/2086)	https://github.com/GSA/px-benefit-finder/issues/2086

<!--- If there are steps for local setup list them here -->
- [ ] create a new criteria

<img width="1112" alt="Screenshot 2025-02-07 at 10 09 17 AM" src="https://github.com/user-attachments/assets/2d11f3b8-a503-4978-9bd7-56bff141b94e" />

- [ ] update the `applicant date of birth` criteria to indicate `Has Child === true` and  `Child Dependency Option === >18years` 

<img width="608" alt="Screenshot 2025-02-11 at 8 57 54 AM" src="https://github.com/user-attachments/assets/7515a181-c1f8-4b5c-84b1-ce200e9d80bf" />

- [ ] edit the life event form for `Benefit finder: disability`
- [ ] under `applicant date of birth` criteria on `/disability`,  add the new child criteria 

<img width="747" alt="Screenshot 2025-02-07 at 10 10 22 AM" src="https://github.com/user-attachments/assets/288985e8-ef65-4514-b4a1-2ca59e346107" />

- [ ] clear cache
- [ ] hard refresh
- [ ] ensure child is nested in the data fetched

<img width="1102" alt="Screenshot 2025-02-07 at 10 12 07 AM" src="https://github.com/user-attachments/assets/09d8f45f-67b6-41b1-89f2-4bf3c316eebc" />

- [ ] navigate to first step
- [ ] confirm child fieldsets are _not_ visible
- [ ] input date: >=18years
- [ ] confirm child fieldsets are visible
- [ ] adjust date: <18years
- [ ] confirm child fieldsets are _not_ visible
- [ ] ensure you can complete form
- [ ] ensure you can validate error alerts
- [ ] ensure expected criteria still validates criteria correctly

<!--- If there are steps for user testing list them here -->

expected:

https://github.com/user-attachments/assets/779eabf6-e529-4209-839a-cff69842926b

[Clicking Back buttons on Firefox browser after entering required fields on ‘death of a loved one’ add an extra route with # on ‘about-the-deceased’ route](https://github.com/GSA/px-benefit-finder/issues/2058)	https://github.com/GSA/px-benefit-finder/issues/2058

- [ ] Open the application in the Firefox browser.
- [ ] Navigate to the ‘Death of a Loved One’ workflow.
- [ ] Fill in all required fields and navigate to results page.
- [ ] Click the browser’s Back button to return to navigate fully back to the Intro component while making note of the URL changes
- [ ] expect Clicking the Back button should navigate back to the ‘about-the-deceased’ route without appending an extra # to the URL.

[child criteria is required cause error](https://github.com/GSA/px-benefit-finder/issues/2089)	https://github.com/GSA/px-benefit-finder/issues/2089

## Detailed Testing steps

<!--- If there are steps for user testing list them here -->
- [ ] under date criteria on `/disability`, create a child criteria with the child dependency value of `>=18years`
- [ ] set criteria to required
- [ ] navigate to first step
- [ ] CLICK next
- [ ] ensure error alert only shows visible required fields
- [ ] input date <18years
- [ ] resolve all other required fields
- [ ] CLICK next
- [ ] ensure step 2 renders
- [ ] CLICK back
- [ ] update date value to >=18years
- [ ] ensure child criteria displays
- [ ] CLICK next
- [ ] ensure child criteria error is in alert
- [ ] resolve error
- [ ] CLICK next
- [ ] ensure step 2 renders
- [ ] CLICK back
- [ ] input date <18years
- [ ] ensure child criteria is not displayed
- [ ] input date >=18years
- [ ] ensure child criteria values are cleared

expected:

https://github.com/user-attachments/assets/b54685f6-e96e-40d7-9acf-5a4afa71fe95

### Custom Module Improvements
[Remove boolean type of criteria](https://github.com/GSA/px-benefit-finder/issues/2080)	https://github.com/GSA/px-benefit-finder/issues/2080

- [ ] Pull changes locally
- [ ] Make local development site up at `http://localhost`
- [ ] Navigate to `admin/content?combine=&type=bears_criteria&status=All&langcode=All`
- [ ] Go to criteria "Applicate date of birth" edit page
- [ ] Verify no Boolean in the list of Type field

<img width="500" src="https://github.com/user-attachments/assets/ff6ee609-9949-4394-b078-733720900236">

[Life event form criteria error message not showing in JSON data](https://github.com/GSA/px-benefit-finder/issues/2088)	https://github.com/GSA/px-benefit-finder/issues/2088

- [ ] Pull changes locally
- [ ] Make local development site up at http://localhost
- [ ] Navigate to `admin/content?combine=&type=bears_life_event_form&status=All&langcode=All`
- [ ] Go to life event form "Benefit finder: disability" edit page
- [ ] Go to criteria "Applicant date of birth"
- [ ] Input "ERROR MESSAGE for applicant date of birth" in Error Message field
- [ ] Save as published
- [ ] Navigate to `benefit-finder/disability`
- [ ] Click `Next` to "About the applicant" page
- [ ] Click `Next`
- [ ] Verify the error message having the value in Error Message field

<img width=600 src="https://github.com/user-attachments/assets/566e299e-178f-4f5f-881f-71bcc9da556f">


[PHPStan error fixing](https://github.com/GSA/px-benefit-finder/issues/2039)	https://github.com/GSA/px-benefit-finder/issues/2039

- [ ] Navigate to node/add/bears_agency to create an agency
- [ ] Language: English
- [ ] Title: agency 1
- [ ] Summary: agency 1 summary
- [ ] Lede: agency 1 lede
- [ ] Save as Published
- [ ] Verify Generating both draft and published JSON data files
- [ ] Navigate to node/add/bears_criteria to create 3 criteria
- [ ] Language: English
- [ ] Title: criteria 1
- [ ] Criteria key: criteria_1
- [ ] ID: criteria_1
- [ ] Type: Radio
- [ ] Label criteria 1 label
- [ ] Name: criteria 1 name
- [ ] Values: 
- [ ] Yes
- [ ] No
- [ ] Check [x] Has Child
- [ ] Child Dependency Option: Yes
- [ ] Save as Published
- [ ] Language: English
- [ ] Title: criteria 2
- [ ] Criteria key: criteria_2
- [ ] ID: criteria_2
- [ ] Type: Radio
- [ ] Label criteria 2 label
- [ ] Name: criteria 2 name
- [ ] Values: 
- [ ] V1
- [ ] V2
- [ ] Check [x] Has Child
- [ ] Child Dependency Option: V1
- [ ] Save as Published
- [ ] Language: English
- [ ] Title: criteria 3
- [ ] Criteria key: criteria_3
- [ ] ID: criteria_3
- [ ] Type: Radio
- [ ] Label criteria 3 label
- [ ] Name: criteria 3 name
- [ ] Values: 
- [ ] W1
- [ ] W2
- [ ] Empty [ ] Has Child
- [ ] Save as Published
- [ ] Navigate to node/add/bears_benefit to create a benefit
- [ ] Language: English
- [ ] Title: benefit 1
- [ ] Headline: benefit 1 headline
- [ ] Summary: benefit 1 summary
- [ ] Life Event Forms - check [x] Benefit finder: death of a loved one
- [ ] Agency: agency 1
- [ ] Source Link: https://www.ssa.gov/
- [ ] Benefit Eligibility
- [ ] Criteria Key: criteria 1
- [ ] Label: criteria 1 label
- [ ] Acceptable values: Yes
- [ ] Criteria Key: criteria 2
- [ ] Label: criteria 2 label
- [ ] Acceptable values: V1
- [ ] Criteria Key: criteria 3
- [ ] Label: criteria 3 label
- [ ] Acceptable values: Yes
- [ ] Save as Published
- [ ] Verify error message

<img width=500 src=https://github.com/user-attachments/assets/eff950f8-97af-4567-953b-92800f80d968>

- [ ] Update criteria 3 Acceptable values: W1
- [ ] Verify Generating both draft and published JSON data files
- [ ] Navigate to admin/content?combine=&type=bears_life_event_form&status=All&langcode=All
- [ ] Go to life event form Benefit finder: death of a loved one edit page
- [ ] In About you section, add 3 criteria
- [ ] Criteria Key: criteria 1
- [ ] Legend: criteria 1 legend
- [ ] Hint: criteria 1 hint
- [ ] Error Message: criteria 1 error
- [ ] Add criteria 2 as criteria 1 child
- [ ] Criteria Key: criteria 2
- [ ] Legend: criteria 2 legend
- [ ] Hint: criteria 2 hint
- [ ] Error Message: criteria 2 error
- [ ] Add criteria 3 as criteria 2 child
- [ ] Criteria Key: criteria 3
- [ ] Legend: criteria 3 legend
- [ ] Hint: criteria 3 hint
- [ ] Error Message: criteria 3 error
- [ ] Save as Published
- [ ] Verify error message

<img width=500 src=https://github.com/user-attachments/assets/bb719720-87d0-418b-9a55-90bead43cbd2>

- [ ] Remove criteria 3
- [ ] Save as Published
- [ ] Verify Generating both draft and published JSON data files
- [ ] Navigate to benefit-finder/api/life-event/death
- [ ] Verify criteria 1 and criteria 2 in JSON data

<img width=100% src=https://github.com/user-attachments/assets/8f7d920c-d4e6-495a-a9ac-ca6452545fa4>

- [ ] Navigate to benefit-finder/death
- [ ] Hard refresh web browser
- [ ] Click Start finding benefits
- [ ] Verify criteria 1 element
- [ ] Select Yes and verify criteria 2 element open
- [ ] Select No and verify criteria 2 element closed

<img width=500 src=https://github.com/user-attachments/assets/d6f95999-0fad-4eb7-a3c1-726cb4fe7ad1>

- [ ] Select criteria 1 Yes
- [ ] Select criteria 2 V1
- [ ] Fill other form elements to get into Get your results page
- [ ] Get into "Benefits you did not qualify for" page 
- [ ] Find the benefit "Benefit 1" to open
- [ ] Verify "Benefit 1" information

<img width=500 src=https://github.com/user-attachments/assets/4b2e2b16-a9d5-4be9-9d58-f0579dd572fc>

- [ ] Navigate to admin/content?combine=&type=bears_life_event_form&status=All&langcode=All
- [ ] Go to life event form "Benefit finder: death of a loved one" edit page
- [ ] Change Time Estimate, for example "4-8 minutes"
- [ ] Save as Draft
- [ ] Verify Generating draft JSON data files

<img width=500 src=https://github.com/user-attachments/assets/187f9c73-b68a-4fcd-8fbd-501bafe5edd1>

- [ ] Navigate to benefit-finder/api/life-event/death?mode=draft
- [ ] Verify updated timeEstimate

<img width=100% src=https://github.com/user-attachments/assets/5444d156-bd1a-46ad-8ba5-bb83982dcd9c>

- [ ] Navigate to benefit-finder/death?mode=draft
- [ ] Hard refresh web browser
- [ ] Verify updated Estimated time

<img width=500 src=https://github.com/user-attachments/assets/7e63409f-af40-49d9-8d7f-bd3da06b38b4>

### a11y Improvements
[Skip to main is breaking on tab enter](https://github.com/GSA/px-benefit-finder/issues/2065)	https://github.com/GSA/px-benefit-finder/issues/2065

Test reset focus, `skip-to-h1` working

- [ ] navigate to `/death`
- [ ] tab into `skip to main` element
- [ ] CLICK enter
- [ ] Expect h1 to be focused
- [ ] CLICK "Next"
- [ ] tab into `skip to main` element
- [ ] CLICK enter
- [ ] Expect h1 to be focused

Test hash anchor in view

- [ ] navigate to `/death`
- [ ] navigate to first step
- [ ] CLICK "Next"
- [ ] CLICK first error
- [ ] expect hash in url to update
- [ ] expect hash to focus

[Announce Web Page Title Changes to Screen Reader Users](https://github.com/GSA/px-benefit-finder/issues/1904)	https://github.com/GSA/px-benefit-finder/issues/1904

- [ ] enable voice over
- [ ] navigate to `/death`
- [ ] expect to hear page title announced
- [ ] navigate through the app, for each route - expect to hear page title announced
- [ ] navigate through the app, for each route - expect to see the page title change

### design Improvements
[Color Updates from USA.gov AUDIT](https://github.com/GSA/px-benefit-finder/issues/2073)	https://github.com/GSA/px-benefit-finder/issues/2073

<!--- If there are steps for user testing list them here -->
- [ ] ensure the following changes have been made

<img width="824" alt="Screenshot 2025-01-29 at 9 52 10 AM" src="https://github.com/user-attachments/assets/05f7d55d-7a12-4f3e-99d8-7d9f18973040" />

### Change Requirements
<!-- Checkboxes to indicate need for changes to some part of the system -->

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review
<!-- Checkboxes to indicate need for review -->

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates


## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

scottqueen-bixal committed:
PHP Stan level 4 updates

scottqueen-bixal committed:
Remove boolean from allowed type

scottqueen-bixal committed:
latest compiled application files

scottqueen-bixal committed:
Merge branch 'dev' into pxbf-release-v0.10.0.beta.1

akf committed:
Merge branch 'dev' into pxbf-release-v0.10.0.beta.1

akf commented:
@scottqueen-bixal I'm hitting a javascript error when I test the nested criteria. I added the nested criteria and added it to the disability Life Event Form as described. Then: 
1. navigate to http://localhost/benefit-finder/disability
2. check json, yes the nested criteria is included
3. click start
4. Enter date of birth January 1, 2000. When I enter the last 0 in "2000" the form disappears and I get these error messages in my console: 

```
benefit-finder.min.js?v=1.x:40 TypeError: Cannot read properties of null (reading '0')
    at Object.$v [as DateEligibility] (benefit-finder.min.js?v=1.x:51:26563)
    at ft (benefit-finder.min.js?v=1.x:78:25518)
    at an (benefit-finder.min.js?v=1.x:78:25779)
    at benefit-finder.min.js?v=1.x:78:25954
    at Array.map (<anonymous>)
    at benefit-finder.min.js?v=1.x:78:25904
    at Array.map (<anonymous>)
    at Ye (benefit-finder.min.js?v=1.x:78:25862)
    at benefit-finder.min.js?v=1.x:78:26081
    at Array.map (<anonymous>)
aa @ benefit-finder.min.js?v=1.x:40
n.callback @ benefit-finder.min.js?v=1.x:40
bu @ benefit-finder.min.js?v=1.x:38
Bu @ benefit-finder.min.js?v=1.x:40
kf @ benefit-finder.min.js?v=1.x:40
bm @ benefit-finder.min.js?v=1.x:40
Pm @ benefit-finder.min.js?v=1.x:40
cn @ benefit-finder.min.js?v=1.x:40
Vu @ benefit-finder.min.js?v=1.x:40
on @ benefit-finder.min.js?v=1.x:38
(anonymous) @ benefit-finder.min.js?v=1.x:40
benefit-finder.min.js?v=1.x:51 Uncaught TypeError: Cannot read properties of null (reading '0')
    at Object.$v [as DateEligibility] (benefit-finder.min.js?v=1.x:51:26563)
    at ft (benefit-finder.min.js?v=1.x:78:25518)
    at an (benefit-finder.min.js?v=1.x:78:25779)
    at benefit-finder.min.js?v=1.x:78:25954
    at Array.map (<anonymous>)
    at benefit-finder.min.js?v=1.x:78:25904
    at Array.map (<anonymous>)
    at Ye (benefit-finder.min.js?v=1.x:78:25862)
    at benefit-finder.min.js?v=1.x:78:26081
    at Array.map (<anonymous>)
$v @ benefit-finder.min.js?v=1.x:51
ft @ benefit-finder.min.js?v=1.x:78
an @ benefit-finder.min.js?v=1.x:78
(anonymous) @ benefit-finder.min.js?v=1.x:78
(anonymous) @ benefit-finder.min.js?v=1.x:78
Ye @ benefit-finder.min.js?v=1.x:78
(anonymous) @ benefit-finder.min.js?v=1.x:78
Op @ benefit-finder.min.js?v=1.x:78
as @ benefit-finder.min.js?v=1.x:38
sa @ benefit-finder.min.js?v=1.x:40
Pf @ benefit-finder.min.js?v=1.x:40
Nf @ benefit-finder.min.js?v=1.x:40
jm @ benefit-finder.min.js?v=1.x:40
Ci @ benefit-finder.min.js?v=1.x:40
wa @ benefit-finder.min.js?v=1.x:40
Vu @ benefit-finder.min.js?v=1.x:40
on @ benefit-finder.min.js?v=1.x:38
(anonymous) @ benefit-finder.min.js?v=1.x:40
```

scottqueen-bixal commented:
> @scottqueen-bixal I'm hitting a javascript error when I test the nested criteria. I added the nested criteria and added it to the disability Life Event Form as described. Then:
> 
> 1. navigate to http://localhost/benefit-finder/disability
> 2. check json, yes the nested criteria is included
> 3. click start
> 4. Enter date of birth January 1, 2000. When I enter the last 0 in "2000" the form disappears and I get these error messages in my console:
> 
> ```
> benefit-finder.min.js?v=1.x:40 TypeError: Cannot read properties of null (reading '0')
>     at Object.$v [as DateEligibility] (benefit-finder.min.js?v=1.x:51:26563)
>     at ft (benefit-finder.min.js?v=1.x:78:25518)
>     at an (benefit-finder.min.js?v=1.x:78:25779)
>     at benefit-finder.min.js?v=1.x:78:25954
>     at Array.map (<anonymous>)
>     at benefit-finder.min.js?v=1.x:78:25904
>     at Array.map (<anonymous>)
>     at Ye (benefit-finder.min.js?v=1.x:78:25862)
>     at benefit-finder.min.js?v=1.x:78:26081
>     at Array.map (<anonymous>)
> aa @ benefit-finder.min.js?v=1.x:40
> n.callback @ benefit-finder.min.js?v=1.x:40
> bu @ benefit-finder.min.js?v=1.x:38
> Bu @ benefit-finder.min.js?v=1.x:40
> kf @ benefit-finder.min.js?v=1.x:40
> bm @ benefit-finder.min.js?v=1.x:40
> Pm @ benefit-finder.min.js?v=1.x:40
> cn @ benefit-finder.min.js?v=1.x:40
> Vu @ benefit-finder.min.js?v=1.x:40
> on @ benefit-finder.min.js?v=1.x:38
> (anonymous) @ benefit-finder.min.js?v=1.x:40
> benefit-finder.min.js?v=1.x:51 Uncaught TypeError: Cannot read properties of null (reading '0')
>     at Object.$v [as DateEligibility] (benefit-finder.min.js?v=1.x:51:26563)
>     at ft (benefit-finder.min.js?v=1.x:78:25518)
>     at an (benefit-finder.min.js?v=1.x:78:25779)
>     at benefit-finder.min.js?v=1.x:78:25954
>     at Array.map (<anonymous>)
>     at benefit-finder.min.js?v=1.x:78:25904
>     at Array.map (<anonymous>)
>     at Ye (benefit-finder.min.js?v=1.x:78:25862)
>     at benefit-finder.min.js?v=1.x:78:26081
>     at Array.map (<anonymous>)
> $v @ benefit-finder.min.js?v=1.x:51
> ft @ benefit-finder.min.js?v=1.x:78
> an @ benefit-finder.min.js?v=1.x:78
> (anonymous) @ benefit-finder.min.js?v=1.x:78
> (anonymous) @ benefit-finder.min.js?v=1.x:78
> Ye @ benefit-finder.min.js?v=1.x:78
> (anonymous) @ benefit-finder.min.js?v=1.x:78
> Op @ benefit-finder.min.js?v=1.x:78
> as @ benefit-finder.min.js?v=1.x:38
> sa @ benefit-finder.min.js?v=1.x:40
> Pf @ benefit-finder.min.js?v=1.x:40
> Nf @ benefit-finder.min.js?v=1.x:40
> jm @ benefit-finder.min.js?v=1.x:40
> Ci @ benefit-finder.min.js?v=1.x:40
> wa @ benefit-finder.min.js?v=1.x:40
> Vu @ benefit-finder.min.js?v=1.x:40
> on @ benefit-finder.min.js?v=1.x:38
> (anonymous) @ benefit-finder.min.js?v=1.x:40
> ```

That is odd, I'll take a look at it this morning.

scottqueen-bixal commented:
I wasn't able to reproduce the error, testing all three (3) criteria types as nested children against the `date of birth` criteria on `disability`.  We may need to triage and dig a bit more.  I'll send you some screen capture footage as well as add more detail to the testing process, incase there was a missing step. 

Here is the fieldset object that was generated in the data when testing,:

```json
  {
      "fieldset": {
          "criteriaKey": "applicant_date_of_birth",
          "legend": "Your date of birth",
          "required": true,
          "hint": "For example: January 19, 2000",
          "errorMessage": "",
          "inputs": [
              {
                  "inputCriteria": {
                      "id": "applicant_date_of_birth",
                      "type": "Date",
                      "name": "applicant_date_of_birth",
                      "label": "Date of birth",
                      "hasChild": true,
                      "childDependencyOption": ">18years",
                      "values": [
                          {
                              "default": "",
                              "value": {}
                          }
                      ]
                  }
              }
          ],
          "children": [
              {
                  "fieldsets": [
                      {
                          "fieldset": {
                              "criteriaKey": "test_nested_child",
                              "legend": "test",
                              "required": true,
                              "hint": "test hint",
                              "errorMessage": "test error message override",
                              "inputs": [
                                  {
                                      "inputCriteria": {
                                          "id": "test_nested_child",
                                          "type": "Radio",
                                          "name": "test nested child",
                                          "label": "test nested child",
                                          "hasChild": false,
                                          "childDependencyOption": "",
                                          "values": [
                                              {
                                                  "option": "yes",
                                                  "value": "yes"
                                              },
                                              {
                                                  "option": "no",
                                                  "value": "no"
                                              }
                                          ]
                                      }
                                  }
                              ],
                              "children": []
                          }
                      }
                  ]
              }
          ]
      }
  },
```

akf committed:
Merge branch 'dev' into pxbf-release-v0.10.0.beta.1

akf reviewed (approved):
Works as described.
